query-engine/query-engine/src/tests/execute_raw.rs

```diff
@@ -341,3 +341,59 @@ async fn other_errors_bubbling_through_to_the_user(api: &TestApi) -> anyhow::Res
 
     Ok(())
 }
+
+#[test_each_connector]
+async fn parameter_escaping(api: &TestApi) -> anyhow::Result<()> {
+    feature_flags::initialize(&[String::from("all")]).unwrap();
+    let query_engine = api.create_engine(&TODO).await?;
+
+    let query = match api.connection_info() {
+        ConnectionInfo::Postgres(_) => {
+            indoc! {r#"
+                mutation {
+                    queryRaw(
+                        query: "SELECT ($1)::text",
+                        parameters: "[\"\\\"name\\\"\"]"
+                    )
+                }
+            "#}
+        }
+        ConnectionInfo::Mssql(_) => {
+            indoc! {r#"
+                mutation {
+                    queryRaw(
+                        query: "SELECT @P1",
+                        parameters: "[\"\\\"name\\\"\"]"
+                    )
+                }
+            "#}
+        }
+        _ => {
+            indoc! {r#"
+                mutation {
+                    queryRaw(
+                        query: "SELECT ?",
+                        parameters: "[\"\\\"name\\\"\"]"
+                    )
+                }
+            "#}
+        }
+    };
+
+    let column_name = match api.connection_info() {
+        ConnectionInfo::Postgres(_) => "text",
+        ConnectionInfo::Mssql(_) => "",
+        _ => "?",
+    };
```

Comment:
tiny nitpick: couldn't this match be avoided with a query like `SELECT ? AS result`?

Reply:
yeah, but copy&pasta